pkg/config.go

```diff
@@ -88,6 +89,25 @@ func (s SystemConfig) DatabaseDriverName() *string {
 	return s.GetValue("databaseDriverName")
 }
 
+// IsSubdomainValid checks if the subdomain is valid.
+// Nil is returned if it is, otherwise an error
+// message describing the violation is returned.
+func (s SystemConfig) IsSubdomainValid(namespace, subdomain string) error {
```

Comment:
This should be in [pkg/util/validate.go](https://github.com/onepanelio/core/blob/master/pkg/util/validate/validate.go). There is already an unused method in there that may be useful.